Obsidian/Serializer/PacketSerializer.cs

```diff
@@ -21,48 +21,57 @@ public static class PacketSerializer
+            try
+            {
+                if (packet == null)
+                    throw new ArgumentNullException(nameof(packet));
 
-            if (stream == null)
-                throw new ArgumentNullException(nameof(stream));
+                if (stream == null)
+                    throw new ArgumentNullException(nameof(stream));
 
-            await stream.Lock.WaitAsync();
+                await stream.Lock.WaitAsync();
 
-            await using var dataStream = new MinecraftStream();
+                await using var dataStream = new MinecraftStream();
 
-            if (packet is ChunkDataPacket chunkData)
-            {
-                await chunkData.WriteAsync(dataStream);
-            }
-            else
-            {
-                var valueDict = packet.GetAllObjects().OrderBy(x => x.Key.Order);
-
-                foreach (var (key, value) in valueDict)
+                if (packet is ChunkDataPacket chunkData)
+                {
+                    await chunkData.WriteAsync(dataStream);
+                }
+                else
                 {
-                    //await Globals.PacketLogger.LogDebugAsync($"Writing value @ {dataStream.Position}: {value} ({value.GetType()})");
+                    var valueDict = packet.GetAllObjects().OrderBy(x => x.Key.Order);
 
-                    var dataType = key.Type;
+                    foreach (var (key, value) in valueDict)
+                    {
+                        //await Globals.PacketLogger.LogDebugAsync($"Writing value @ {dataStream.Position}: {value} ({value.GetType()})");
 
-                    if (dataType == DataType.Auto)
-                        dataType = value.GetType().ToDataType();
+                        var dataType = key.Type;
 
-                    await dataStream.WriteAsync(dataType, key, value);
-                }
-            }
+                        if (dataType == DataType.Auto)
+                            dataType = value.GetType().ToDataType();
 
-            var packetLength = packet.Id.GetVarIntLength() + (int)dataStream.Length;
+                        await dataStream.WriteAsync(dataType, key, value);
+                    }
+                }
 
-            await stream.WriteVarIntAsync(packetLength);
-            await stream.WriteVarIntAsync(packet.Id);
+                var packetLength = packet.Id.GetVarIntLength() + (int)dataStream.Length;
 
-            dataStream.Position = 0;
-            // await dataStream.DumpAsync(packet: packet);
+                await stream.WriteVarIntAsync(packetLength);
+                await stream.WriteVarIntAsync(packet.Id);
 
-            await dataStream.CopyToAsync(stream);
+                dataStream.Position = 0;
+                // await dataStream.DumpAsync(packet: packet);
 
-            stream.Lock.Release();
+                await dataStream.CopyToAsync(stream);
+            }
+            catch
+            {
+                throw;
+			}
```

Comment:
Formatting mistake

Reply:
I've made the exception logging off by default + fixed the invalid formatting in PacketSerializer.cs